iree/tools/iree-benchmark-module-main.cc

```diff
@@ -165,6 +105,136 @@ void RegisterModuleBenchmarks(
       // we can make this setting configurable with a custom command line flag.
       ->Unit(benchmark::kMillisecond);
 }
+
+Status GetModuleContentsFromFlags(std::string& module_data) {
+  auto module_file = absl::GetFlag(FLAGS_module_file);
+  IREE_ASSIGN_OR_RETURN(module_data, file_io::GetFileContents(module_file));
+  return iree::OkStatus();
+}
+
+// TODO(hanchung): Consider to refactor this out and reuse in iree-run-module.
+// This class helps organize required resources for IREE. The order of
+// construction and destruction for resources matters. And the lifetime of
+// resources also matters. The lifetime of IREEBenchmark should be as long as
+// ::benchmark::RunSpecifiedBenchmarks() where the resources are used during
+// benchmarking.
+class IREEBenchmark {
+ public:
+  IREEBenchmark()
+      : instance_(nullptr),
+        device_(nullptr),
+        hal_module_(nullptr),
+        context_(nullptr),
+        input_module_(nullptr){};
+  ~IREEBenchmark() {
+    // Order matters. This is caller's responsibility to clean up the resources.
```

Comment:
No longer caller's responsibility?

```suggestion
    // Order matters.
```

Reply:
I intended to say those should be released by the class itself. I think since we are using pointer here, it does have context that the class needs to release it. Remove that comment.